.github/workflows/build-tests.yml

```
@@ -0,0 +1,51 @@
+name: Build-Testing
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```

Trigger workflow only on file changes in `IOT` folder. See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-paths

```suggestion
    branches: [ main ]
    paths: [ 'IOT/**' ]
```